### DISCUSSION TITLE
ondata not invoked when the results between calls don't change

### DISCUSSION THREAD

**Discussion by danielvanmil** [apollo-elements/apollo-elements#279] [Q&A]
Hello,

With my query, for example:

```
  private readonly getItems = new ApolloQueryController(this, GetItemsDocument, {
    onData: () => {
     console.log("onData");
    },
    onError: () => {
      console.error("Cannot get data.");
    },
  });
```

The onData is not invoked when a next query result is the same as the previous one (in my case no items).
This seems expected behavior as the Appollo ObservableQuery code contains a check with a isDifferentFromLastResult:

```
    ObservableQuery.prototype.reportResult = function (result, variables) {
        var lastError = this.getLastError();
        if (lastError || this.isDifferentFromLastResult(result)) {
            if (lastError || !result.partial || this.options.returnPartialData) {
                this.updateLastResult(result, variables);
            }
            iterateObserversSafely(this.observers, 'next', result);
        }
    };
```
- Is this correct?
- Should this be documented as I would expect the onData handler to be invoked always?

Thanks.

**Comment by bennypowers** [comment]
Seems to me this is the correct behavior, that function should only be invoked when the data actually changes

**Reply by danielvanmil** [reply]
ok, but, should it then be called onDataChange maybe and how can I respond after the loading of the controller if needed?

**Reply by bennypowers** [reply]
you could dirty check in the host's updated callback

**Reply by danielvanmil** [reply]
@bennypowers : the dirty check seems a good solution/workaround for my issue. But it still seems to me a bit of doc could be added to make it more clear. Thanks!

**Reply by bennypowers** [reply]
you could send a docs PR
